### PR TITLE
Updates supported Python versions and uses tox

### PR DESCRIPTION
**Pull request by funkyfuture**
i was curious whether i could use the library with recent Python versions and ended up with this.

### CODE REVIEW

**Comment by Changaco** [comment]
Why did you remove the `__future__` imports?

**Comment by Changaco** [comment]
Oh, you commented that they're "unneeded". That may be so but I think it's safer to leave them as they were.

**Comment by funkyfuture** [comment]
what would be safer by remaining unused code? 

if one will be needed (from those four, only `absolut_import` *may* make sense at some point), the tests will fail.

**Comment by Changaco** [comment]
Removing `unicode_literals` changes the type of every literal string in the file under python2. For all I know that could lead to unicode errors that aren't detected by the tests, even if it seems unlikely.

**Comment by Changaco** [comment]
In a stable project like `dependency_injection` it's best not to make cosmetic changes that might break stuff, and in any project that supports both python 2 and 3 having all the `__future__` imports is common practice even if they're not all strictly necessary.

**Comment by funkyfuture** [comment]
imo, this is not a cosmetic change. it removes code that - again imo - makes no sense and just introduces overhead.

there are neither print statements, absolute imports nor divisions in the code.

the only string literals in the code are used in the string-representation of an exception and as parameters to `namedtuple`. Under Python2, the former couldn't be unicode (it's a non-unicode literal that formats callables' names that cannot be unicode) and for the latter unicode characters are not allowed at all.

i don't get the concern.

**Comment by whit537** [comment]
@funkyfuture Are any of the changes on this PR hard requirements for the tests to pass on Python > 3.3?

**Comment by funkyfuture** [comment]
sorry for the late reply. nope, code changes are necessary for flake8. the `__future__` imports removal is just caused by my obsession to remove anything that isn't needed.

feel free to pick what you find useful.

**Comment by whit537** [comment]
Gotcha, thanks. In that case, I'm going to go ahead and close this out. I would merge a minimal PR that updates the versions in `.travis.yml`, but the rest of it is more than I want to think through right now.

**Comment by Changaco** [comment]
Rebased on master and amended the commit that removed the `__future__` imports.

**Comment by Changaco** [comment]
Tests are passing, I'm merging.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM funkyfuture/nest-of-serpents
```
Comment (whit537) [inline Dockerfile]: What happens when @funkyfuture changes this? Also, security hole?

Comment (funkyfuture) [inline Dockerfile]: just add a note that @funkyfuture might behave nasty at some point and runnng the tests with that script is on everyone's own risk?

Comment (Changaco) [inline Dockerfile]: This stuff is only for running tests, and in an environment that's supposed to be isolated from the rest of your system, so the security risk is very low, probably not even worth a warning.

**Pull request merged**